src/tree-internal.component.ts

```
@@ -147,13 +147,18 @@ export class TreeInternalComponent implements OnInit, OnChanges, OnDestroy, Afte
         while (i--) {
           const node = e.captured[i];
           const ctrl = this.treeService.getController(node.tree.id);
-          if (ctrl.isChecked()) {
+          if (ctrl && ctrl.isChecked()) {
```

Under what conditions would ctrl be undefined? By the looks of it, it looks like we're hiding potential problems by silently ignoring conditions where ctrl is not defined.

---

I think originally this was caused by an issue where moving a checked parent with children, would destroy the parent tree and all its child trees, which in turn would delete the controllers for all of them and then re-create them all in the new tree/location, so the child controllers were unavailable for the rest of the loop after the parent was removed. This was fixed subsequently, so I have removed this guard now.